scripts/pylib/twister/twisterlib/runner.py

```diff
@@ -1666,8 +1666,8 @@ def cmake_assemble_args(extra_args, handler, extra_conf_files, extra_overlay_con
                             extra_dtc_overlay_files, cmake_extra_args,
                             build_dir):
         # Retain quotes around config options
-        config_options = [arg for arg in extra_args if arg.startswith("CONFIG_")]
-        args = [arg for arg in extra_args if not arg.startswith("CONFIG_")]
+        config_options = [arg for arg in extra_args if arg.startswith(("CONFIG_", "SB_CONFIG_"))]
```

Comment:
maybe better would be to add `_CONFIG_` to enable configuration of sysbuild images,
like `mcuboot_CONFIG_`, `<network_core>_CONFIG`

Reply:
That's a good point. Only doing this for these 2 won't cover all cases, and doing e.g. `hci_net_CONFIG_MY_STR_VAL="a"` would still fail with this

Reply:
> maybe better would be to add _CONFIG_ to enable configuration of sysbuild images,
like mcuboot_CONFIG_, <network_core>_CONFIG

That would be nice but would be a different / bigger re-write task.
`startswith()` only checks from start or the optional start position (not used here).
But we don't know the start position for images as names may vary in length.

Also we don't want to match everything with `_CONFIG_` in it, cause that would end up matching CMake settings like `APPLICATION_CONFIG_DIR`.

So for now, I prefer to ensure that `SB_CONFIG_` is handled in addition to `CONFIG_`, as handling of `<image>_CONFIG_<setting>` is a very different task and might involve some a valid / known image list type to avoid matching cases like above.

Reply:
Could it be something like startswith("CONFIG" or "SB_CONFIG" or "[<lower_case>_]+_CONFIG")? Then it wouldn't apply to APPLICATION_CONFIG_DIR unless they name their Kconfig option application_CONFIG_DIR

Reply:
except that there is no requirement for a sample to be all lower case.

I fully acknowledge current PR is only a minor improvement and not the final solution, but I still find it an beneficial improvement until someone has time to improve twister with a bigger re-work.